File: DataLayer/Entities/DetectionResultImage.cs
using DataLayer.Entities.Common;

namespace DataLayer.Entities
{
    public class DetectionResultImage : EntityBase, IImage
    {
        public int DetectionId { get; set; }
        public int DetectionResultId { get; set; }
        public DetectionResult DetectionResult { get; set; }
        public string Name { get; set; }
        public string Thumbnail { get; set; }
        public string Url { get; set; }

        public string GetPath()
        {
            return $"{nameof(DetectionResultImage)}/{DetectionId}";
        }
    }
}

File: dataLayer/entities/detection_result_image.py
from sqlalchemy import Column, String, Integer, Date, ForeignKey

from dataLayer.database_connection import Base


class DetectionResultImage(Base):
    __tablename__ = 'DetectionResultImage'

    id = Column(Integer, primary_key=True)
    name = Column('Name', String)
    thumbnail = Column("Thumbnail", String)
    detection_id = Column("DetectionId", Integer, ForeignKey("Detection.Id"))

    def __init__(self, name, detection_id):
        self.name = name
        self.detection_id = detection_id
